### PR TITLE
trivial: prepare itest for `blockbeat`

### PR DESCRIPTION
**Pull request by yyforyongyu**
Cherry-picked commits from the final `blockbeat` to reduce the PR size.

This PR makes a few trivial changes to fix the flakes found in the blockbeat itest.

### CODE REVIEW

File: lntest/harness_node_manager.go [hunk 1]
```diff
@@ -134,6 +134,9 @@ func (nm *nodeManager) shutdownNode(node *node.HarnessNode) error {
 func (nm *nodeManager) restartNode(ctxt context.Context,
 	hn *node.HarnessNode, callback func() error) error {
 
+	// Make sure to register the node after it has been restarted.
+	defer nm.registerNode(hn)
```
Comment (guggero) [inline lntest/harness_node_manager.go:138]: Just out of curiosity: Why is this needed? It doesn't look like we're calling into `shutdownNode()` here, which is what actually removes the node from the list of active nodes.

**Review by guggero — APPROVED** [review]
Awesome work tracking down all those flakes :100:  LGTM :tada:

File: lntest/harness_node_manager.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/harness_node_manager.go:138]: yeah seems unnecessary - let me remove it and run the CI in the other branch again.

Comment (yyforyongyu) [reply lntest/harness_node_manager.go]: you are right - think it's a mistake when debugging flakes, now removed!

File: lntest/harness.go [hunk 2]
```diff
@@ -2302,7 +2306,6 @@ func (h *HarnessTest) openChannelsForNodes(nodes []*node.HarnessNode,
 	p OpenChannelParams) []*lnrpc.ChannelPoint {
 
 	// Sanity check the params.
-	require.False(h, p.ZeroConf, "zero-conf channels must be disabled")
```
Comment (ViktorTigerstrom) [inline lntest/harness.go:2305]: Ah my bad! Sorry about that suggestion :/!

**Review by ViktorTigerstrom — APPROVED** [review]
LGTM :rocket::fire:!
